app/models/spree/preferences/store.rb

```diff
@@ -36,11 +36,10 @@ def get(key, fallback = nil)
           return val
         end
 
-        # If it's not in the cache, maybe it's in the database, but
-        # has been cleared from the cache
+        # If it's not in the cache, maybe it's in the database, but has been cleared from the cache
         # does it exist in the database?
-        if should_persist? && (Spree::Preference.table_exists? &&
-          preference = Spree::Preference.find_by(key: key))
+        preference = Spree::Preference.find_by(key: key)
+        if should_persist? && (Spree::Preference.table_exists? && preference)
```

Comment:
This could blow up on certain commands. You're trying to read from the table before checking if it exists.